File: golang/services/service.go
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package services

import "net"

/*
The identifier used for services with the network.
*/
type ServiceID string

/*
Importan info related to the service.
*/
type ServiceInfo struct {
	IPAddress net.IP
}

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
type Service interface {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	IsAvailable() bool
}

File: typescript/src/lib/services/service.ts
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

/*
The identifier used for services with the network.
*/
export type ServiceID = string;

export class SharedDirectory {
    private readonly absFilepathOnThisContainer: string;
    private readonly absFilepathOnServiceContainer: string;

    constructor (absFilepathOnThisContainer: string, absFilepathOnServiceContainer: string) {
        this.absFilepathOnThisContainer = absFilepathOnThisContainer;
        this.absFilepathOnServiceContainer = absFilepathOnServiceContainer;
    }

    public getAbsFilepathOnThisContainer(): string {
        return this.absFilepathOnThisContainer;
    }

    public getAbsFilepathOnServiceContainer(): string {
        return this.absFilepathOnServiceContainer;
    }
}
